File: server/php/src/Infrastructure/Mappers/PaymentDetails/AmountOfMoneyMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\AmountOfMoney as AmountOfMoneyDto;
use OnlinePayments\Sdk\Domain\AmountOfMoney as AmountOfMoneySdk;

final class AmountOfMoneyMapper
{
    public static function mapFromSdkResponse(?AmountOfMoneySdk $response): ?AmountOfMoneyDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new AmountOfMoneyDto();
        $dto->amount = $response?->amount ?? null;
        $dto->currencyCode = $response?->currencyCode ?? null;

        return $dto;
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/AmountOfMoneyMapper.cs
using AmountOfMoneySdk = OnlinePayments.Sdk.Domain.AmountOfMoney;
using AmountOfMoneyDto = Business.Domain.Payments.PaymentDetails.AmountOfMoney;

namespace Infrastructure.Mappers.PaymentDetails;

public static class AmountOfMoneyMapper
{
    public static AmountOfMoneyDto Map(AmountOfMoneySdk? response)
    {
        return new()
        {
            Amount = response?.Amount,
            CurrencyCode = response?.CurrencyCode,
        };
    }
}